README.md

```diff
@@ -74,3 +74,67 @@ is as follows
+
+### Supported Endpoints
+You can provide a filter as a request parameter for `GET` endpoints that return multiple resources. The filter expression should be assigned to a request parameter named `filter`.  
+Examples of supported endpoints for providing a filter parameter are as follows:
+- Getting multiple resources endpoint - `http://localhost:5000/resources/artists`
+- Getting multiple related resources for a parent resource - `http://localhost:5000/resources/concerts/37e813a2-bb28-11ec-8422-0242ac120001/artists`
+
+
+### Forming Expressions
+Filter expressions can be formed through the use of logical operators or SpEL method invocation. The expression requires an operator and 1 or 2 operands depending on the type of operator.
+
+#### Logical Operators
+When forming a filter expression using logical operators the following conditions must be met:
+- The left side operand must be a reference to a field on the entity. Literal values are not supported here.
+- The right side operand can be a field reference or a literal value, however the data type of the right  
+  side operand must match the data type of the left side operand.
+- The following operators are supported: `==, !=, >=, >, <=, <`
+- The operators can also be provided in the equivalent textual form such as: `eq, ne, ge, gt, le, lt`
+
+An example of the filter request parameter using a logical operator is as follows:
+`filter=id == 'dc27d5aa-4e7d-474e-98b6-ebf9aae9a471'`
+
+#### Method References
+When forming a filter expression using SpEL method invocation the following conditions must be met:
+- Expressions can be formed using either of the following (case-insensitive) method references: `IN, BETWEEN`
+- There should be at least 2 parameters passed to the method depending on which one it is.
+- The first parameter (LHS) must be a reference to a field on the entity. Literal values are not supported here.
+- The parameter/s (RHS) after the first must be literal value/s and their data types must match that of the first parameter.
+  - For the  `BETWEEN` method reference there must be exactly two values after the first param.
+  - For the `IN` method reference there must be at least 1 value after the first param.
+
+An example of the filter request parameter using method references is as follows:
+`filter=in(id, '27e813a2-bb28-11ec-8422-0242ac120001', '27e813a2-bb28-11ec-8422-0242ac120003')`
+
+
+### Combining Conditions
+There are a few additional logical operators that can be applied and also used to build a filter expression made up of multiple conditions.
+- This can be applied with any of the following operators: `&&, ||, !`
+- The operators can also be provided in the equivalent textual form such as: `and, or, not`
+
+An example of the filter request parameter using additional logical operators is as follows:
+`filter=!in(id, '27e813a2-bb28-11ec-8422-0242ac120001', '27e813a2-bb28-11ec-8422-0242ac120003')`
```

Comment:
Can you combine filters? E.g. startDate gt X and endDate lt Y

If so an example would be really helpful.

Reply:
indeed you can. have now added an example